### DISCUSSION TITLE
Ending sequence issues: batteries running out and submarine cannot progress without drone

### DISCUSSION THREAD

**Discussion by Vykn** [FakeFishGames/Barotrauma#14384] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

A few times on the discord players have expressed that they do not know where to go in the ending sequence - specifically that they cannot find the button that opens the door on EndOutpost1. Seemingly, some players are taking so long in the level such that the battery in the sonar beacon is running out (it takes 16 minutes for it to run out), and so the sonar target for the location of the button disappears. It's not super common, but would be an easy fix by making the contained battery invulnerable.

Secondly, and more severely, if a submarine with a drone drops the drone and tries to progress, they will be not be taken to the next level in the sequence. They will either need to go back and collect the drone or restart the level. Only occurs if the drone was brought into and dropped inside the ending level.

### Reproduction steps

1. Teleport to the end of the campaign with a submarine with a drone (tested with kastrull)
2. Wait around for 17 minutes and observe that the "Emergency Transmission" sonar target disappears
3. Drop the drone somewhere away from the exit point (assuming that it needs to be >50m away to not count as a "leaving sub") 
4. open the gate
5. Drive into the anomaly and observe that the sub is not taken to the next level

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report! I unfortunately couldn't find a way to reproduce the issue with the drone, I seem to be able to enter the portal with no issues despite the drone being nowhere near it:

https://github.com/user-attachments/assets/eee25cfd-7a9b-4d9f-97a7-4e298d63a05a

**Reply by Vykn** [reply]
Just tested again and it only seems to occur when the drone is inside the structure:


https://github.com/user-attachments/assets/93fbd2f3-90cb-47df-ac81-c4e510cb3537
